### PR TITLE
Fix/#786 column headers to histogram output of abundance-dist.py

### LINKED ISSUE #786: documentation of .hist output file
The .hist output file of abundance-dist.py needs to be documented. I think the four columns of the file refer to  "k-mer counts", "Frequency", "cumulative frequency" , and "cumulative fraction"

### PR DESCRIPTION
**Pull request by drtamermansour**
Fix #786 : adding column headers to the histogram output of abundance-dist.py **as an option**.
The column headers of this output file are already documented in doc/user/scripts.txt.

### CODE REVIEW

**Comment by drtamermansour** [comment]
make doc passes but make test gives me an Error
"tests.test_version.test_python_and_c_match_base ... FAIL"
Any suggestions?

**Comment by ctb** [comment]
Yes, 'make clean test'.  But that's not the problem with Jenkins - it looks like there's a pep8 issue, diagnose with 'make pep8'.

File: scripts/abundance-dist.py [hunk 1]
```diff
@@ -47,6 +47,9 @@ def get_parser():
                         + khmer.__version__)
     parser.add_argument('-f', '--force', default=False, action='store_true',
                         help='Overwrite output file if it exists')
+    parser.add_argument('--Header-names', dest='Header_names', default=False,
```
Comment (ctb) [inline scripts/abundance-dist.py]: I think this should be '--header-names'  - it's rare that long-form arguments have capital letters in them, for whatever reason.

File: scripts/abundance-dist.py [hunk 2]
```diff
@@ -95,6 +98,10 @@ def main():
         sys.exit(1)
     hash_fp = open(args.output_histogram_filename, 'w')
 
+    if args.Header_names:
+        print("k-mer_abundance", "K-mer_count", "Cumulative_count" ,
```
Comment (ctb) [inline scripts/abundance-dist.py]: Hmm, this is python3.0 format for print, I think? Didn't realize it worked in python 2.7 :).  Should be

print >> hash_fp, ...

to be in line with the rest of the codebase. Not sure how important this is?

**Comment by ctb** [comment]
pep8 issue:

```
pep8 --exclude=_version.py setup.py khmer/ scripts/ tests/ || true
scripts/abundance-dist.py:102:67: E203 whitespace before ','
scripts/abundance-dist.py:102:69: W291 trailing whitespace
```

File: scripts/abundance-dist.py (hunk 2 above)
Comment (drtamermansour) [inline scripts/abundance-dist.py]: The same print function is used already in the same script !
https://github.com/ged-lab/khmer/blob/master/scripts/abundance-dist.py#L106

**Comment by drtamermansour** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by drtamermansour** [comment]
ready for review

**Comment by mr-c** [comment]
Replaces by #831

**Pull request closed without merging**